packages/api/src/router/incident.ts

```diff
@@ -16,37 +22,249 @@ export const incidentRouter = createTRPCRouter({
+
+  updateIncidentUpdate: protectedProcedure
+    .input(insertIncidentUpdateSchema)
+    .mutation(async (opts) => {
+      const currentWorkspace = await opts.ctx.db
+        .select()
+        .from(workspace)
+        .where(eq(workspace.slug, opts.input.workspaceSlug))
+        .get();
+      const currentUser = opts.ctx.db
+        .select()
+        .from(user)
+        .where(eq(user.tenantId, opts.ctx.auth.userId))
+        .as("currentUser");
+      const result = await opts.ctx.db
+        .select()
+        .from(usersToWorkspaces)
+        .where(eq(usersToWorkspaces.workspaceId, currentWorkspace.id))
+        .innerJoin(currentUser, eq(usersToWorkspaces.userId, currentUser.id))
+        .get();
+
+      if (!result) return;
+
+      const { workspaceSlug, ...incidentUpdateInput } = opts.input;
+
+      if (!incidentUpdateInput.id) return;
+
+      const currentIncident = await opts.ctx.db
+        .update(incidentUpdate)
+        .set(incidentUpdateInput)
+        .where(eq(incidentUpdate.id, incidentUpdateInput.id))
+        .returning()
+        .get();
+
+      return currentIncident;
+    }),
+
+  deleteIncident: protectedProcedure
+    .input(z.object({ id: z.number() }))
+    .mutation(async (opts) => {
+      // TODO: this looks not very affective
+      const currentUser = await opts.ctx.db
+        .select()
+        .from(user)
+        .where(eq(user.tenantId, opts.ctx.auth.userId))
+        .get();
+
+      const result = await opts.ctx.db
+        .select()
+        .from(usersToWorkspaces)
+        .where(eq(usersToWorkspaces.userId, currentUser.id))
+        .all();
+
+      const workspaceIds = result.map((workspace) => workspace.workspaceId);
+
+      const incidentToDelete = await opts.ctx.db
+        .select()
+        .from(incident)
+        .where(
+          and(
+            eq(incident.id, opts.input.id),
+            inArray(incident.workspaceId, workspaceIds),
+          ),
+        )
+        .get();
+      if (!incidentToDelete) return;
+
+      await opts.ctx.db
+        .delete(incident)
+        .where(eq(incident.id, incidentToDelete.id))
+        .run();
+    }),
+
+  deleteIncidentUpdate: protectedProcedure
+    .input(z.object({ id: z.number() }))
+    .mutation(async (opts) => {
+      const currentUser = await opts.ctx.db
+        .select()
+        .from(user)
+        .where(eq(user.tenantId, opts.ctx.auth.userId))
+        .get();
+
+      const result = await opts.ctx.db
+        .select()
+        .from(usersToWorkspaces)
+        .where(eq(usersToWorkspaces.userId, currentUser.id))
+        .all();
+
+      const workspaceIds = result.map((workspace) => workspace.workspaceId);
+
+      const incidentUpdateToDelete = await opts.ctx.db
+        .select()
+        .from(incidentUpdate)
+        .where(and(eq(incidentUpdate.id, opts.input.id)))
+        // FIXME: check if incident related to workspaceId
+        // .innerJoin(incident, inArray(incident.workspaceId, workspaceIds))
```

Comment:
Would be nice to check if user is allowed to delete the `incidentUpdate` - but we are not storing the `workspaceId`. Needs some table joining.